torch/distributed/distributed_c10d.py

```diff
@@ -1554,19 +1564,20 @@ def all_gather_multigpu(
         work.wait()
 
 
-def _object_to_tensor(obj):
+def _object_to_tensor(obj, device):
     f = io.BytesIO()
     _pickler(f).dump(obj)
     byte_storage = torch.ByteStorage.from_buffer(f.getvalue())  # type: ignore[attr-defined]
     # Do not replace `torch.ByteTensor` or `torch.LongTensor` with torch.tensor and specifying dtype.
     # Otherwise, it will casue 100X slowdown.
     # See: https://github.com/pytorch/pytorch/issues/65696
-    byte_tensor = torch.ByteTensor(byte_storage)
-    local_size = torch.LongTensor([byte_tensor.numel()])
+    byte_tensor = torch.ByteTensor(byte_storage).to(device)
```

Comment:
This reminds me, we should probably figure out the root cause issue of https://github.com/pytorch/pytorch/issues/65696, because we really don't want to be using `ByteTensor` and `LongTensor` style APIs anymore.